scenes/teleporter/teleporter.gd

```diff
@@ -29,21 +29,46 @@ func _ready() -> void:
 	set_collision_mask_value(1, true)
 
 	if Engine.is_editor_hint():
+		_update_available_spawn_points()
+		notify_property_list_changed()
 		return
 
 	self.body_entered.connect(_on_body_entered, CONNECT_ONE_SHOT)
 
 
 func _on_body_entered(_body: PhysicsBody2D) -> void:
-	if scene_to_go_to:
+	if scene_to_go_to and scene_to_go_to != get_tree().current_scene.scene_file_path:
 		# We are using call_deferred here because removing nodes with
 		# collisions during a callback caused by a collision might cause
 		# undesired behavior.
 		SceneSwitcher.change_to_file_with_transition.call_deferred(scene_to_go_to, spawn_point_path)
+	else:
+		var spawn_point: SpawnPoint = get_node_or_null(spawn_point_path)
+		if is_instance_valid(spawn_point):
+			spawn_point.move_player_to_self_position(true)
+			self.body_entered.connect(_on_body_entered, CONNECT_ONE_SHOT)
+
+
+func _scene_to_go_to_path():
```

Comment:
I think that is better to prefix this function with "get", as it's obtaining the path. And the return hint would be even better:
```suggestion
func _get_scene_to_go_to_path() -> String:
```